### PR TITLE
aeprimeop typing

### PR DESCRIPTION
**Pull request by rustaem**
Lots of dialyzer annotations, one (I believe) critical bug (in aeprimop.run_contract/), couple less important dismatches.

### CODE REVIEW

**Comment by cytadela8** [comment]
Rebase!!!

**Comment by cytadela8** [comment]
Ok, so now that I can see what's fixed here. Seems you only looked into aeprimop. You can rebase maybe on "master" branch and let's merge this there. :P Also one comment.

File: apps/aeprimop/src/aeprimop_state.erl [hunk 1]
```diff
@@ -7,143 +7,199 @@
...
-        , put_oracle_query/2
-        ]).
-
--export([ cache_write_through/1
-        , set_var/4
-        , tx_env/1
-        ]).
+-export([delete_x/3
+    , delete_account/2
+    , delete_contract/2
```
Comment (cytadela8) [inline apps/aeprimop/src/aeprimop_state.erl:12]: Please leave old indentation style, it's prefered.

**Comment by rustaem** [comment]
> Ok, so now that I can see what's fixed here. Seems you only looked into aeprimop. You can rebase maybe on "master" branch and let's merge this there. :P Also one comment.

Will fix. Please do not accept it yet: that critical bug is already fixed in 3618, and I think I've found some minor issues to be considered before accepting.

File: apps/aeprimop/src/aeprimop.erl [hunk 2]
```diff
@@ -1603,7 +1775,13 @@ contract_call_fail(Call0, Fee, S) ->
     {Account, S3} = get_account(Payer, S2),
     account_spend(Account, UsedAmount, S3).
 
+<<<<<<< HEAD
```
Comment (cytadela8) [inline apps/aeprimop/src/aeprimop.erl:1778]: Rebase error. Please be carefull when reabasing... Git tells you when a file has issues. There are multiple ways to deal with them. Personally I frequenetly use `git mergetool`

**Review by cytadela8 — CHANGES_REQUESTED** [review]
1. Make sure the code compiles
2. Make sure tests pass
3. Ping me for review

**Review by cytadela8 — DISMISSED** [review]
Good!

To whoever pushes the merge button: I highly suggest using "squash and merge" to remove the commit mess.

File: apps/aeprimop/src/aeprimop.erl [hunk 3]
```diff
@@ -2120,19 +2307,20 @@ assert_contract_call_stack(CallStack, S) ->
...
     end.
 
+-spec assert_auth_call_object_not_exist(aect_call:call(), state()) -> ok | no_return().
 assert_auth_call_object_not_exist(Call, S) ->
     AuthCallId = aect_call:id(Call),
     Pubkey     = aect_call:caller_pubkey(Call),
     case find_auth_call(Pubkey, AuthCallId, S) of
         none -> ok;
-        {value, _} -> runtime_error(auth_call_object_already_exist)
+        {_, _} -> runtime_error(auth_call_object_already_exist)
```
Comment (hanssv) [inline apps/aeprimop/src/aeprimop.erl:2316]: Maybe just `_`?

File: apps/aeprimop/src/aeprimop_state.erl [hunk 4]
```diff
@@ -54,96 +54,153 @@
...
-    #state{ trees = Trees
-          , cache = dict:new()
-          , env   = dict:new()
-          , height = aetx_env:height(TxEnv)
-          , tx_env = TxEnv
-          , protocol = aetx_env:consensus_version(TxEnv)
-          }.
-
+    #state{trees = Trees
+        , cache = dict:new()
```
Comment (hanssv) [inline apps/aeprimop/src/aeprimop_state.erl:105]: Bad indentation

File: apps/aeprimop/src/aeprimop_state.erl [hunk 5]
```diff
@@ -400,44 +482,46 @@ cache_write_through_fun({contract, Pubkey}, Contract, Trees) ->
...
 %%%===================================================================
 %%% Variable environment
 
+-spec get_var(channel_key(), _, state()) -> any().
 get_var({var, X}, Tag, #state{env = E}) when is_atom(X) ->
     {Tag, Val} = dict:fetch(X, E),
     Val;
 get_var({X, Y} = Res, oracle_query, #state{}) when is_binary(X),
-                                                   is_binary(Y) ->
+    is_binary(Y) ->
```
Comment (hanssv) [inline apps/aeprimop/src/aeprimop_state.erl:514]: This isn't easier to read? Rather the opposite.

**Review by hanssv — COMMENTED** [review]
Great effort!

Some nit-picking and some grumpy comments. We don't have an agreed upon code style - so feel free to write new modules in a style of your choice (within reason) but changing existing code (unless correcting obvious errors) is not a contribution. It will make git blaming harder, etc.

**Comment by rustaem** [comment]
> Great effort!
> 
> Some nit-picking

Thanks, @hanssv ! I beg to differ: you're not nit-picking. This is a great piece of tech, so aesthetic is important, I'm aiming for it. So if you feel anything's wrong, I just revert the changes with a due "thank you". Please check the new revision when you have time.

File: apps/aeprimop/src/aeprimop.erl (hunk 3 above)
Comment (rustaem) [reply apps/aeprimop/src/aeprimop.erl:2316]: '{_, _}' is consistent through functions here. Are you sure '_' pattern won't break things? Yes, I tested and it seems fine, but may be there is some obscure case in vivo?

**Review by hanssv — APPROVED** [review]
👍

**Review by cytadela8 — APPROVED** [review]

**Pull request merged**